csharp/src/Google.Protobuf/Collections/RepeatedField.cs

```diff
@@ -34,8 +34,8 @@ public sealed class RepeatedField<T> : IList<T>, IList, IDeepCloneable<RepeatedF
         private static readonly T[] EmptyArray = new T[0];
         private const int MinArraySize = 8;
 
-        private T[] array = EmptyArray;
-        private int count = 0;
+        internal T[] array = EmptyArray;
```

Comment:
Please don't make fields internal. I was already nervous about having the SetCount method, but I *definitely* don't want internal access to fields.

Reply:
Instead, I'd recommend adding internal SetCount and AsSpan methods, and calling those from the public methods in UnsafeCollectionOperations.

Reply:
Yes, I understand. I follow the same patterns as in

https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/List.cs#L25-L27

It appears to be a performance optimization in the `List<T>` class to expose these fields. 

I can revert this change.